### PR TITLE
PR: Change IPython Console icon

### PR DESCRIPTION
**Pull request by juanis2112**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)
* [x] Included a screenshot (if affecting the UI)

I changed IPython Console icon.

screen shot 2019-02-14 at 12 30 04 am

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

Juanis2112

### CODE REVIEW

File: spyder/plugins/ipythonconsole/plugin.py [hunk 1]
```diff
@@ -388,6 +388,15 @@ def get_plugin_actions(self):
             return client.get_options_menu()
         return self.menu_actions
 
+    def get_current_tab_manager(self):
+        """Return the plugin as tab_manager"""
```
Comment (CAM-Gerlach) [inline spyder/plugins/ipythonconsole/plugin.py]: Missing period.

File: spyder/plugins/ipythonconsole/widgets/client.py [hunk 2]
```diff
@@ -135,6 +135,8 @@ def __init__(self, plugin, id_,
         self.allow_rename = True
         self.stderr_dir = None
         self.is_error_shown = False
+        self.filename = self.given_name
```
[the remark this answers is not in the document]
Comment (CAM-Gerlach) [reply spyder/plugins/ipythonconsole/widgets/client.py]: More broadly, we should rename the ``File Switcher`` to just the ``Switcher``, or to something else more representative of its expanded functionality (``Focus Switcher``? ``Quick Nav``? What do other IDEs call this?)

**Comment by ccordoba12** [comment]
@juanis2112, please remove the last two commits in this PR and merge it with master, so we can have it for beta2.

We really need to create a proper API for the fileswitcher to show the open IPython consoles there, but @goanpeca is going to take care of that because it's not an easy task.

File: spyder/app/mainwindow.py [hunk 3]
```diff
@@ -3060,9 +3060,11 @@ def add_to_fileswitcher(self, plugin, tabs, data, icon):
             self.fileswitcher = FileSwitcher(self, plugin, tabs, data, icon)
         else:
             self.fileswitcher.add_plugin(plugin, tabs, data, icon)
-
-        self.fileswitcher.sig_goto_file.connect(
+        try:
+            self.fileswitcher.sig_goto_file.connect(
                 plugin.get_current_tab_manager().set_stack_index)
+        except:
+            pass
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: @juanis2112, please revert this change.

File: spyder/plugins/ipythonconsole/plugin.py [hunk 4]
```diff
@@ -405,6 +405,8 @@ def register_plugin(self):
                                      self.set_current_client_working_directory)
 
         self.tabwidget.currentChanged.connect(self.update_working_directory)
+        self.main.add_to_fileswitcher(self, self.tabwidget, self.clients,
+                                      ima.icon('ipython_console'))
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/plugin.py]: And also revert this change.

File: spyder/app/mainwindow.py (hunk 3 above)
Comment (CAM-Gerlach) [reply spyder/app/mainwindow.py]: Also, as a general note, you should essentially *never* use a bare except (equivalent to ``except BaseException:``, vs, at the very least, ``except Exception:``), as the former incorrectly catch ``SystemExit``, ``KeyboardInterrupt`` and ``GeneratorExit`` and prevents Spyder from being correctly closed by these if happens to be in the block in question (or gets stuck in it, etc.). 

Furthermore, if you do ``except Exception:``, you should think very carefully about what purpose it serves and include a comment justifying it over something more specific and tailored, as it will also hide errors from our tests and error reporting system, silently retaining potentially fixable problems.

File: spyder/utils/icon_manager.py [hunk 5]
```diff
@@ -162,8 +162,8 @@
     'error':                   [('fa.times-circle',), {'color': 'darkred'}],
     'warning':                 [('fa.warning',), {'color': 'orange'}],
     'todo':                    [('fa.exclamation',), {'color': '#3775a9'}],
-    'ipython_console':         [('spyder.ipython-logo-alt',), {'color': MAIN_FG_COLOR}],
-    'ipython_console_t':       [('spyder.ipython-logo-alt',), {'color':'gray'}],
+    'ipython_console':         [('mdi.console',), {'color': MAIN_FG_COLOR}],
+    'ipython_console_t':       [('mdi.console',), {'color':'gray'}],
```
Comment (CAM-Gerlach) [inline spyder/utils/icon_manager.py]: Fix PEP 8 warning

```suggestion
    'ipython_console_t':       [('mdi.console',), {'color': 'gray'}],
```

**Comment by ccordoba12** [comment]
@CAM-Gerlach, please see the new title of this PR. Now it's very simple, it just wants to change the IPython console icon, but @juanis2112 left some things that I asked her to remove.

Instead of wasting your time and ours testing this, please help me to decide if PR #8734 is ready or not.

**Comment by CAM-Gerlach** [comment]
@ccordoba12 New title or not, it still says it fixes #8519 which includes connecting consoles to the file switcher, and your only comment that actually said it was being descoped was almost two months old and gave the reason as making it in for Beta 2, which has already been released, so it wasn't 100% clear.

Therefore, given the unclear situation, severity of the problem and the fact that the CIs weren't detecting it, it was not an unreasonable decision to post a short comment pointing it out, as if there was even a small chance it related to desired functionality in this PR, the potential consequences outweighed the mere few seconds it "wasted" of your time.

> Instead of wasting your time and ours testing this, please help me to decide if PR #8734 is ready or not.

That's *exactly* what I *was* doing, until you chose to waste your time (far more than the few seconds you spent reading my comment) writing up a whole reply criticizing me for it, and now requiring me to waste mine to respond to your needless accusations.

**Comment by ccordoba12** [comment]
I modified the PR description. Look, when I'll require your help, I'll ask for it. Otherwise, please don't make me argue about small details.

**Comment by CAM-Gerlach** [comment]
I will follow this up in a more appropriate venue, namely private Gitter chat.

**Comment by CAM-Gerlach** [comment]
For what its worth, I agree with your requested icon change, though.

File: spyder/app/mainwindow.py (hunk 3 above)
Comment (ccordoba12) [reply spyder/app/mainwindow.py]: @juanis2112, what I meant by "revert" is that you need to leave things as they were before. This means removing the try/except you added here, so this change doesn't appear as part of this PR.

File: spyder/plugins/ipythonconsole/plugin.py (hunk 4 above)
Comment (ccordoba12) [reply spyder/plugins/ipythonconsole/plugin.py]: And please remove these lines because they are not needed anymore.

File: spyder/utils/icon_manager.py (hunk 5 above)
Comment (ccordoba12) [reply spyder/utils/icon_manager.py]: To address this, please press the button `Commit suggestion` above and after that `Commit changes`.
